pkg/operator/rules.go

```diff
@@ -36,15 +36,6 @@ import (
 	namespacelabeler "github.com/prometheus-operator/prometheus-operator/pkg/namespacelabeler"
 )
 
-func init() {
-	// For now, the operator only supports legacy label names.
-	// Eventually the operator should support UTF-8 label names too and the
-	// issue is tracked by
-	// https://github.com/prometheus-operator/prometheus-operator/issues/7362
-	// nolint:staticcheck
-	model.NameValidationScheme = model.LegacyValidation
```

Comment:
hmm it still has an impact on relabeling configurations, no?

Reply:
We are updating relabel config here https://github.com/prometheus-operator/prometheus-operator/pull/7637/files#diff-aca3d056c83da1ddc448c8c40960960d7ed00ef61f8d5f08a40e165564d74dcdR322

Reply:
I meant that the LabelName type still has kubebuilder validation restricting to the legacy validation.